### PR TITLE
UCT/UCP: Fix compilation errors with armcc-22.1

### PR DESCRIPTION
**Pull request by brminich**
## What
Fix compilation errors with armclang-22.1

### CODE REVIEW

**Review by rakhmets — DISMISSED** [review]

File: src/ucp/wireup/ep_match.c [hunk 1]
```diff
@@ -88,7 +88,7 @@ ucp_ep_h ucp_ep_match_retrieve(ucp_worker_h worker, uint64_t dest_uuid,
                                ucp_ep_match_conn_sn_t conn_sn,
                                ucs_conn_match_queue_type_t conn_queue_type)
 {
-    ucp_ep_flags_t exp_ep_flags = UCP_EP_FLAG_ON_MATCH_CTX;
+    ucp_ep_flags_t UCS_V_UNUSED exp_ep_flags = UCP_EP_FLAG_ON_MATCH_CTX;
```
Comment (yosefe) [inline src/ucp/wireup/ep_match.c:91]: replace the usage of exp_ep_flags by the flag

File: src/uct/ib/base/ib_verbs.h [hunk 2]
```diff
@@ -71,6 +71,9 @@ static inline ucs_status_t uct_ib_query_device(struct ibv_context *ctx,
 #  define IBV_ACCESS_RELAXED_ORDERING               0
 #endif
 
+#if !HAVE_DECL_IBV_ACCESS_ON_DEMAND
+#  define IBV_ACCESS_ON_DEMAND                      0
+#endif
```
Comment (yosefe) [inline src/uct/ib/base/ib_verbs.h:76]: why needed? can't we remove it from the code?

**Review by yosefe — COMMENTED** [review]
can we add arm22 to ci?

File: src/uct/ib/base/ib_verbs.h (hunk 2 above)
Comment (brminich) [reply src/uct/ib/base/ib_verbs.h:76]: Can you pls elaborate?
Avoiding compilation failure with arm clang on specific config

File: src/ucp/wireup/ep_match.c (hunk 1 above)
Comment (brminich) [reply src/ucp/wireup/ep_match.c:91]: you mean rename to `flags`? (note we may add a couple of flags to this var, not just one)

File: src/uct/ib/base/ib_verbs.h (hunk 2 above)
Comment (yosefe) [reply src/uct/ib/base/ib_verbs.h:76]: AFAIR, we removed IBV_ACCESS_ON_DEMAND because it's supposed to be unused. Why do we need to return it?

Comment (yosefe) [reply src/uct/ib/base/ib_verbs.h:76]: right..
so, maybe we need to expose from UCT that ODP is disabled if IBV_ACCESS_ON_DEMAND==0?
even if it's supported by FW

File: src/ucp/wireup/ep_match.c (hunk 1 above)
Comment (yosefe) [reply src/ucp/wireup/ep_match.c:91]: I mean remove this variable at all. and use UCP_EP_FLAG_ON_MATCH_CTX constant instead of it

Comment (brminich) [reply src/ucp/wireup/ep_match.c:91]: we also add `UCP_EP_FLAG_REMOTE_ID` to the flags var below

Comment (yosefe) [reply src/ucp/wireup/ep_match.c:91]: ok

**Comment by artemry-nv** [comment]
> can we add arm22 to ci?

@Al3xR01 FYI

**Review by yosefe — DISMISSED** [review]
pls squash

**Comment by Al3xR01** [comment]
>

**Comment by Al3xR01** [comment]
We can add arm22 to ci once we add arm platform support.

**Review by yosefe — APPROVED** [review]

**Comment by Al3xR01** [comment]
arm22 in CI: addressed by PR #8895
(https://github.com/openucx/ucx/pull/8895/files#diff-940fa8ef89b586de941ddb47d8566f84686adbcfe592e0d2ab28ef96d7bf4f73)

**Pull request merged**